programs/ui-wrapper/src/processors/settle_funds.rs

```diff
@@ -0,0 +1,189 @@
+use std::{cell::RefMut, str::FromStr};
+
+use borsh::{BorshDeserialize, BorshSerialize};
+use hypertree::{get_mut_helper, DataIndex, RBNode};
+use manifest::{
+    program::{get_mut_dynamic_account, withdraw_instruction},
+    quantities::{QuoteAtoms, WrapperU64},
+    state::{DynamicAccount, MarketFixed},
+    validation::{ManifestAccountInfo, Program, Signer},
+};
+use solana_program::{
+    account_info::{next_account_info, AccountInfo},
+    entrypoint::ProgramResult,
+    instruction::{AccountMeta, Instruction},
+    program::invoke,
+    program_error::ProgramError,
+    pubkey::Pubkey,
+};
+
+use crate::{market_info::MarketInfo, wrapper_state::ManifestWrapperStateFixed};
+
+use super::shared::{
+    check_signer, get_market_info_index_for_market, sync_fast, WrapperStateAccountInfo,
+};
+
+const FEE_DENOMINATOR: u128 = 10u128.pow(9);
+
+#[derive(BorshDeserialize, BorshSerialize, Clone)]
+pub struct WrapperSettleFundsParams {
+    fee_mantissa: u32,
+    platform_fee_percent: u8,
+}
+impl WrapperSettleFundsParams {
+    pub fn new(fee_mantissa: u32, platform_fee_percent: u8) -> Self {
+        WrapperSettleFundsParams {
+            fee_mantissa,
+            platform_fee_percent,
+        }
+    }
+}
+
+pub(crate) fn process_settle_funds(
+    _program_id: &Pubkey,
+    accounts: &[AccountInfo],
+    data: &[u8],
+) -> ProgramResult {
+    let account_iter: &mut std::slice::Iter<AccountInfo> = &mut accounts.iter();
+    let wrapper_state: WrapperStateAccountInfo =
+        WrapperStateAccountInfo::new(next_account_info(account_iter)?)?;
+    let owner: Signer = Signer::new(next_account_info(account_iter)?)?;
+    let trader_token_account_base: &AccountInfo = next_account_info(account_iter)?;
+    let trader_token_account_quote: &AccountInfo = next_account_info(account_iter)?;
+    let market: ManifestAccountInfo<MarketFixed> =
+        ManifestAccountInfo::<MarketFixed>::new(next_account_info(account_iter)?)?;
+    let vault_base: &AccountInfo = next_account_info(account_iter)?;
+    let vault_quote: &AccountInfo = next_account_info(account_iter)?;
+    let mint_base: &AccountInfo = next_account_info(account_iter)?;
+    let mint_quote: &AccountInfo = next_account_info(account_iter)?;
+    let executor_program: Program = Program::new(
```

Comment:
is there a more descriptive name for this?

Reply:
not yet, can figure out a name on monday